torch/lib/libshm/core.cpp

```diff
@@ -46,14 +46,13 @@ void start_manager() {
   }
   SYSCHECK_ERR_RETURN_NEG1(close(pipe_ends[1]));
 
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  ssize_t bytes_read;
-  // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-magic-numbers,cppcoreguidelines-avoid-c-arrays)
-  char buffer[1000];
+  ssize_t bytes_read = 0;
+  constexpr auto MAX_BUFFER_SIZE = 1000;
+  std::array<char, MAX_BUFFER_SIZE> buffer{};
   std::string handle;
   for (;;) {
-    SYSCHECK_ERR_RETURN_NEG1(bytes_read = read(pipe_ends[0], buffer, sizeof(buffer)));
-    handle.append(buffer, bytes_read);
+    SYSCHECK_ERR_RETURN_NEG1(bytes_read = read(pipe_ends[0], buffer.data(), buffer.size()));
+    handle.append(buffer.data(), bytes_read);
     if (bytes_read == 0 || handle[handle.length() - 1] == '\n') {
```

Comment:
Nit (and imo, it would be better to replace `for(;;)` with `while(handle.empty() || handle.back() != '\n')`
```suggestion
    if (bytes_read == 0 || handle.back() == '\n') {
```